### PR TITLE
Print cli logs as json if flag is set

### PR DESCRIPTION
**Pull request by rishflab**

### CODE REVIEW

**Comment by rishflab** [comment]
@binarybaron Could you test this to see if it does what you were asking. I will merge it if it works.

Thanks

**Comment by binarybaron** [comment]
> @binarybaron Could you give this a shot to see if it does what you were asking. I will merge it if it works.
> 
> Thanks

Thank you, two small issues.

1. When a swap with `--json` the terminal printer doesn't print JSON logs but normally formatted logs. 
```
./swap buy-xmr --receive-address **** --seller /dnsaddr/unstoppableswap.net/p2p/12D3KooWLHbHtkK53WoBA82Yht3PCCx6KXYxoJ6QqgXg1rSNtWhR --change-address *** --testnet
Connected to Alice at /dnsaddr/unstoppableswap.net/p2p/12D3KooWLHbHtkK53WoBA82Yht3PCCx6KXYxoJ6QqgXg1rSNtWhR
Received quote price=0.00392300 BTC minimum_amount=0.00010000 BTC maximum_amount=0.10000000 BTC
```

2. The `tracing_appender` does not format the logs it writes to the log file using JSON. I think this can be changed by initializing the layer with `json()`
```
let appender = tracing_appender::rolling::never(dir, format!("swap-{}.log", swap_id));
        let (appender, guard) = tracing_appender::non_blocking(appender);
        std::mem::forget(guard);
        let file_logger = fmt::layer()
            .with_ansi(false)
            .with_target(false)
            .json()
            .with_writer(appender);
```

File: swap/src/cli/tracing.rs [hunk 1]
```diff
@@ -88,6 +96,19 @@ fn debug_terminal_printer<S>() -> StdErrPrinter<StdErrLayer<S, ChronoLocal>> {
     }
 }
 
+fn debug_json_terminal_printer<S>() -> StdErrPrinter<StdErrJsonLayer<S, ChronoLocal>> {
+    let is_terminal = atty::is(atty::Stream::Stderr);
+    StdErrPrinter {
+        inner: fmt::layer()
+            .with_ansi(is_terminal)
+            .with_target(false)
```
Comment (da-kami) [inline swap/src/cli/tracing.rs:104]: The `info_json_terminal_printer` includes `.with_level(false)` - here you don't. Is that on purpose?

**Review by da-kami — APPROVED** [review]
left one comment, don't think it's critical

File: swap/src/cli/tracing.rs (hunk 1 above)
Comment (rishflab) [reply swap/src/cli/tracing.rs:104]: yes on purpose. keeping it consistent with the non json printer.

**Pull request merged**